src/components/activity-history-list/utils.tsx

```diff
@@ -16,155 +16,176 @@ const {
-                return (
-                    <div key={index}>
-                        <p>
-                            <b>{activitiesOnTargetType[paramName].field}</b>
-                        </p>
-                        <p>
-                            {previouslyLabel}{' '}
-                            <b>
-                                {activitiesOnTargetType[paramName].output(
-                                    oldData[paramName]
-                                )}
-                            </b>
-                        </p>
-                        <p>
-                            {changedToLabel}{' '}
-                            <b>
-                                {activitiesOnTargetType[paramName].output(
-                                    newData[paramName]
-                                )}
-                            </b>
-                        </p>
-                    </div>
-                );
-            }
+    if (targetType === 'contactDetails') {
+        return (
+            <div>
+                <p>
+                    <b>{contactType(newData.contactType)}</b>
+                </p>
+                <p>
+                    {addedLabel} <b>{newData.value}</b>
+                </p>
+            </div>
         );
-
-        return updatedElements;
     }
 
-    if (type === 'delete') {
-        if (targetType === 'contactDetails') {
-            return (
-                <>
-                    <div>
-                        <p>
-                            <b>{contactType(oldData.contactType)}</b>
-                        </p>
-                        <p>
-                            {removedLabel} <b>{oldData.value}</b>
-                        </p>
-                    </div>
-                </>
-            );
-        }
-
-        const removedData = Object.keys(oldData).filter((paramName: string) => {
-            if (oldData[paramName] === newData[paramName]) return;
-            return oldData[paramName];
-        });
+    if (targetType === 'tenure') {
+        return (
+            <p>
+                <b>{entityCreated(targetType)}</b>
+            </p>
+        );
+    }
+};
 
-        return removedData.map((paramName: string, index) => {
-            if (paramName === 'id') return;
+export const UpdatedEntityRecord = ({
+    targetType,
+    newData,
+    oldData,
+}: ActivityChangeRecord): any => {
+    const activitiesOnTargetType = activities[targetType];
+    const updatedElements = Object.keys(newData).map(
+        (paramName: string, index) => {
+            if (
+                paramName === 'id' ||
+                activitiesOnTargetType[paramName] === undefined
+            )
+                return null;
+
+            if (oldData[paramName] === newData[paramName]) return null;
             return (
                 <div key={index}>
                     <p>
                         <b>{activitiesOnTargetType[paramName].field}</b>
```

Comment:
If the field is not in locale, can we default to the paramName? Or is that handled elsewhere?

Reply:
Implemented case to default to paramName if field is not in locale